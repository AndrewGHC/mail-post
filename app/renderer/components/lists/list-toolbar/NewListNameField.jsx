import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import { TextField } from 'redux-form-material-ui';

import { validateName } from './new-list-helpers';

const NewListNameField = ({
  name,
  disabled,
  lists,
  ...rest
}) => (
  <Field
    name={name}
    component={TextField}
    placeholder="List name"
    disabled={disabled}
    validate={(...args) => validateName(lists, ...args)}
    {...rest}
  />
);

NewListNameField.propTypes = {
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool.isRequired,
  lists: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default NewListNameField;
