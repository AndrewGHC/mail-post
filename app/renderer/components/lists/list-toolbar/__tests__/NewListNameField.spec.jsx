import React from 'react';
import { shallow } from 'enzyme';

import NewListNameField from '../NewListNameField';

const props = {
  name: 'howdy',
  disabled: false,
  anotherProp: 'hi',
  lists: [
    {
      id: 1,
      name: 'hi',
    },
    {
      id: 2,
      name: 'ho',
    },
    {
      id: 1,
      name: 'hum',
    },
  ],
};
describe('NewListNameField', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<NewListNameField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes props', () => {
    const wrapper = shallow(<NewListNameField {...props} />);
    expect(wrapper.prop('name')).toEqual(props.name);
    expect(wrapper.prop('disabled')).toEqual(props.disabled);
    expect(wrapper.prop('anotherProp')).toEqual(props.anotherProp);
  });
});
