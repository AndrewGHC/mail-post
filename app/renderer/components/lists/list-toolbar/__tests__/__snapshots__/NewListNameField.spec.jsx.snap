// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewListNameField matches snapshot 1`] = `
<Field
  anotherProp="hi"
  component={[Function]}
  disabled={false}
  name="howdy"
  placeholder="List name"
  validate={[Function]}
/>
`;
